src/rtcModule/webrtc.cpp

```diff
@@ -334,41 +336,48 @@ void RtcModule::handleCallData(Chat &chat, Id chatid, Id userid, uint32_t client
         return;
     }
 
-    if (state == Call::CallDataState::kCallDataEnd)
+    //Compatibility
+    if (state == Call::CallDataState::kCallDataRinging || state == Call::CallDataState::kCallDataSessionKeepRinging)
+    {
+        ringing = true;
+    }
+    else if (state == Call::CallDataState::kCallDataEnd)
     {
         // Peer will be removed from call participants with OP_ENDCALL
         return;
     }
 
-    updatePeerAvState(chatid, callid, userid, clientid, avFlagsRemote);
-    if (state == Call::CallDataState::kCallDataRinging ||
-            state == Call::CallDataState::kCallDataSessionKeepRinging)
-    {
-        handleCallDataRequest(chat, userid, clientid, callid, avFlagsRemote);
-        if (state == Call::CallDataState::kCallDataSessionKeepRinging)
-        {
-            auto itCallHandler = mCallHandlers.find(chatid);
-            if (itCallHandler != mCallHandlers.end() && !itCallHandler->second->getInitialTimeStamp())
-            {
-                itCallHandler->second->setInitialTimeStamp(time(NULL));
-            }
-        }
-    }
-    else if (state == Call::CallDataState::kCallDataNotRinging)
+    auto itCall = mCalls.find(chatid);
+    if (itCall != mCalls.end())
     {
-        auto itCall = mCalls.find(chatid);
-        if (itCall == mCalls.end())
+        if (itCall->second->id() != callid)
         {
-            RTCM_LOG_DEBUG("Ingoring kCallDataNotRinging CALLDATA for unknown call");
+            RTCM_LOG_WARNING("Ignoring CALLDATA because its callid is different than the call that we have in that chatroom");
             return;
         }
 
-        if (chat.isGroup() && itCall->second->state() == Call::kStateRingIn)
+        updatePeerAvState(chatid, callid, userid, clientid, avFlagsRemote);
+
+        if (itCall->second->state() == Call::kStateRingIn && itCall->second->isCaller(userid, clientid) && !ringing)
         {
+            if (!chat.isGroup())
+            {
+                RTCM_LOG_WARNING("Received not-terminat CALLDATA with ringing flag oof for 1on1 in kStateRingIn. "
+                                 "The call should have already beein takien out of this state by a RTMSG");
+            }
+
             itCall->second->destroy(TermCode::kAnswerTimeout, false);
         }
+
+        itCall->second->updateAvFlags(userid, clientid, avFlagsRemote);
```

Comment:
Isn't it already done in line 359?? --> `updatePeerAvState(chatid, callid, userid, clientid, avFlagsRemote);`